app/src/main/java/ar/com/wolox/android/example/utils/DynamicScrollRV.kt

```diff
@@ -0,0 +1,31 @@
+package ar.com.wolox.android.example.utils
+
+import androidx.recyclerview.widget.RecyclerView
+
+abstract class DynamicScrollRV(val performAction: () -> Unit) : RecyclerView.OnScrollListener() {
+
+    var currentScroll = 0
+
+    private val TAG: String = DynamicScrollRV::class.java.simpleName
+
+    override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+        super.onScrollStateChanged(recyclerView, newState)
+        if (newState == RecyclerView.SCROLL_STATE_IDLE && currentScroll >= 0 && !isLoading() && hasMore()) performAction()
```

Comment:
Which is the meaning of `0`? It's a magic number? Please, move it to companion object